pillar.example

```diff
@@ -19,3 +19,57 @@ docker:
     all: true # true (Default): Remove all unused images, translates to `docker system prune --force --all`
               # false: (Remove only dangling images, translates to `docker system prune --force`
     volumes: false # Remove unused volumes as well. Default: false
+
+
+# Registry credentials for the docker.login state (if used)
+#
+# Note that multiple dictionaries, and multiple registries per dictionary, can be specified:
+#
+
+docker-registries:
+  'https://docker-registry.work.com':
+    username: jrabbit
+    password: s3cr3t
+
+my-private-docker-registries:
+  'https://docker-A.jack-rabbit.com':
+    username: jack
+    password: c4rr0t
+  'https://docker-B.jack-rabbit.com':
+    username: jack
+    password: c4rr0t
+
+
+# Configuration for the docker.machine substate
+#
+# See README on how to obtain the contents of the key and certificate files as well as valid versions
+#
+# TL;DR versions: https://gitlab.com/gitlab-org/ci-cd/docker-machine/-/releases
+#
+# TL;DR certs:
+# - docker-machine create --driver <DRIVER> <OPTIONS> init-vm
+# - wait until init-vm reports as ready
+# - extract required data from /root/.docker/machine/certs
+# - docker-machine rm init-vm
+#
+# The files ca-key.pem and key.pem contain sensitive information. It might be advisable to use Salt's
+# GPG renderer feature to encrypt the content.
+#
+docker-machine:
+  version: v0.16.2-gitlab.18
```

Comment:
This bothers me a little bit. Maybe we can put a comment here, that users are responsible of upgrading the file here? 

This is hard to automate within Saltstack, but users could be using some sort of script if they like. But I think just a hint that this is used for updates and updates should be taken care of would be good I think, to not encourage users to stay on ancient (potentially insecure) versions.

Reply:
updated both the pillar.example TL;DR and a slightly more elaborate remark in the README